internal/consensus/state.go

```diff
@@ -2348,124 +2400,122 @@ func (cs *State) addVote(vote *types.Vote, peerID p2p.ID) (added bool, err error
 		// TODO punish a peer if it sent a vote with an extension when the feature
 		// is disabled on the network.
 		// https://github.com/tendermint/tendermint/issues/8565
-		return false, fmt.Errorf("received vote with vote extension for height %v (extensions disabled) from peer ID %s", vote.Height, peerID)
+		return fmt.Errorf("received vote with vote extension for height %v (extensions disabled) from peer ID %s", vote.Height, peerID)
 	}
+	return nil
+}
 
-	height := cs.Height
-	added, err = cs.Votes.AddVote(vote, peerID, extEnabled)
-	if !added {
-		// Either duplicate, or error upon cs.Votes.AddByIndex()
-
-		// If the vote wasn't added but there's no error, its a duplicate vote
-		if err == nil {
-			cs.metrics.DuplicateVote.Add(1)
-		}
-		return added, err
+// handleDuplicateVote handles a duplicate vote.
+func (cs *State) handleDuplicateVote(err error) {
+	// If the vote wasn't added but there's no error, it's a duplicate vote
+	if err == nil {
+		cs.metrics.DuplicateVote.Add(1)
 	}
+}
+
+// updateMetricsAndEvents updates metrics and events based on the vote.
+func (cs *State) updateMetricsAndEvents(vote *types.Vote) {
 	if vote.Round == cs.Round {
 		vals := cs.state.Validators
 		_, val := vals.GetByIndex(vote.ValidatorIndex)
 		cs.metrics.MarkVoteReceived(vote.Type, val.VotingPower, vals.TotalVotingPower())
 	}
 
 	if err := cs.eventBus.PublishEventVote(types.EventDataVote{Vote: vote}); err != nil {
-		return added, err
+		cs.Logger.Error("failed to publish event vote", "err", err)
```

Comment:
Are we ignoring this error now?

Reply:
will check that.

Reply:
@melekes it's safe to say, that if we are ignoring that error, it should be found by a test, right?

Would it make sense to address this by adding a test?

Reply:
we definitely are.  Nice catch!  Fixing...